### PR TITLE
Testing feature

### PR DESCRIPTION
**Pull request by SaveryIV**
In this project we have added:

- Test cases for all the components.
- Tests for the logic components.

### CODE REVIEW

**Review by joel-kalema — COMMENTED** [review]
Hi TEAM

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request  🛑 

pirates-of-the-caribbean-johnny-depp


- [ ] You have submitted a project that doesn't have a pull request description


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @joel-kalema in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/Home.test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import '@testing-library/jest-dom/jest-globals';
+import Home from './Home';
+
+describe('Testing Home', () => {
+});
+test('Render Home component', () => {
+  const { getByTestId } = render(<Home />);
+  const element = getByTestId('display');
+  expect(element).toBeInTheDocument();
+});
```
Comment (DeliceLydia) [inline src/components/Home.test.js]: - [ ] Please use jest `snapshot` to all components

File: src/components/Calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    fireEvent.click(getByText('2'));
+    expect(displayElement.textContent).toBe('1+2');
+
+    fireEvent.click(getByText('='));
+    expect(displayElement.textContent).toBe('3');
+
+    fireEvent.click(getByText('AC'));
+    expect(displayElement.textContent).toBe('');
+  });
+});
```
Comment (DeliceLydia) [inline src/components/Calculator.test.js]: - [ ] Please use jest `snapshot` to all components

File: src/components/Button.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+});
+
+test('calls onClick when button is clicked', () => {
+  const onClickMock = jest.fn();
+  const { getByText } = render(<CalculatorButton label="AC" onClick={onClickMock} />);
+  const buttonElement = getByText('AC');
+
+  fireEvent.click(buttonElement);
+  expect(onClickMock).toHaveBeenCalled();
+});
```
Comment (DeliceLydia) [inline src/components/Button.test.js]: - [ ] Please use jest `snapshot` to all components

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Tested all components
- Created git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Alusp — APPROVED** [review]
Hi @SaveryIV and @Zuwairanajma , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights

- [x] Write unit tests for the files operate.js and calculate.js using Jest Successfully ✔️
- [x] Use Jest snapshots to test the components Successfully ✔️
- [x] All tests passing correctly.  ✔️


## Optional suggestions

- [OPTIONAL]  Please ensure that your `Quotes` button returns the required data on the UI as it is throwing errors at the moment.

image


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alusp in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**